Dev 863 passkey docs (#58)
customgento/docs: pages/passkey-m2.md

---
title: Passkey (Magento 2)
permalink: passkey-m2.html

> summary: "The Passkey Extension for Magento 2, enables administrators to log in using a passkey"

Reviewer: ```suggestion
summary: "The Passkey Extension for Magento 2 enables administrators to log in using a passkey"
```

sidebar: pk-m2-sidebar
toc: false
---

> ## Description

Reviewer: We need to somehow explain what exactly a passkey is and link somewhere, where it is explained, why this approach is secure.
Maybe we do this here?

> The Passkey Extension for Magento 2, enables administrators to log in using a passkey. It enhances the user experience

Reviewer: ```suggestion
The Passkey Extension for Magento 2 enables administrators to log in using a passkey. It enhances the user experience
```

by simplifying the login process while maintaining high-security standards. This is ideal for shop administrators who
wish to log in without a password while still ensuring security.

#### Key Features:

> 1. **Passkey:** By Passkey you can log in without using your username and password.

Reviewer: Calling a key feature of a module named passkey passkey, is kind of strange in my eyes.
Maye we can call it "Easy Login" or something similar?

> 2. **Secure:** The Passkey is a secure way to log in without using your username and password.

Reviewer: "The passkey" sounds strange here in my eyes, not like a proper English sentence. Please try to rephrase

> 3. **Easy to Use:** The Passkey is easy to use and can be used by anyone.

Reviewer: Same again, maybe "The Passkey Module", or "CustomGento_Passkey" ... ?

4. **Customizable:** It is possible to allow only specific admin users to use the Passkey.
5. **Multi-Device Support:** Login with different devices like a mobile phone, tablet, or Yubi-Key.

## Requirements

- ext-curl: `*`,
- magento/framework `^103.0`
- magento/module-authorization `^100.4`
- magento/module-backend `^102.0`
- magento/module-store `^101.1`
- magento/module-two-factor-auth `^1.1`
- magento/module-user `^101.2`
- PHP `~8.1.0||~8.2.0||~8.3.0`

## Compatibility

- Magento Open Source / Adobe Commerce >= `2.4`

## Installation Instructions

You can install the extension via Composer or by copying the code into your Magento installation.

### Composer Installation

> 1. `composer require customgento/passkey`

Reviewer: as far as I know, it's `customgento/module-passkey`, right?

2. `bin/magento module:enable CustomGento_Passkey`
3. `bin/magento setup:upgrade`
4. `bin/magento setup:di:compile`
5. `bin/magento cache:flush`

### Manual Installation

1. unzip the downloaded files
2. create the directory `app/code/CustomGento/Passkey/`: `mkdir -p app/code/CustomGento/Passkey/`
3. copy the unzipped files to the newly created directory `app/code/CustomGento/Passkey/`
4. `bin/magento module:enable CustomGento_Passkey`
5. `bin/magento setup:upgrade`
6. `bin/magento setup:di:compile`
7. `bin/magento cache:flush`

## Configuration
You can enable the Passkey feature in the Magento backend under `Stores > Configuration > Security > 2FA > General`.

> It is important to mention that the Passkey can not be the only 2FA method enabled. You need to enable at least one other 2FA method.

Reviewer: ```suggestion
It is important to mention that Passkey can not be the only 2FA method enabled. You need to enable at least one other 2FA method.
```

Reviewer: That is also still ope, right?

If you just choose Passkey as the only 2FA method, you will get an error message.

![Settings configuration]({{ "images/passkey/passkey-configuration.png" }} "Settings configuration")

## Usage

> After enabling the Passkey, you need to register your Passkey device, it is possible to register different devices like a mobile phone or a tablet or Yubi-Key. for registering a new device, you need to first log out, and then try to log in with your username and password again.

Reviewer: ```suggestion
After enabling Passkey, you need to register your Passkey device, it is possible to register different devices like a mobile phone or a tablet or Yubi-Key. for registering a new device, you need to first log out, and then try to log in with your username and password again.
```

> After enabling the Passkey, you need to register your Passkey device, it is possible to register different devices like a mobile phone or a tablet or Yubi-Key. for registering a new device, you need to first log out, and then try to log in with your username and password again.

Reviewer: Please check again all the commas here. It looks like a huge sentence, with thousands of commas.
Please better make shorter sentences, that's easier to read and understand

![Passkey device registration Email]({{ "images/passkey/passkey-device-registration-email.png" }} "Passkey device registration Email")

> Then, You will be asked to check your email and click on the link you received. After clicking on the link, you will be asked to register your Passkey device. You can then choose the device you want to register and follow the instructions.

Reviewer: You will be asked, you will be asked...
Could we find a better wording instead of repeating it?

> Then, You will be asked to check your email and click on the link you received. After clicking on the link, you will be asked to register your Passkey device. You can then choose the device you want to register and follow the instructions.

Reviewer: Could you extend a bit, what "follow the instructions" means at this point?

### Biometric Protection

Passkeys often use biometric authentication (fingerprint, face recognition) or device-based PINs. These are harder to replicate and never transmitted, making unauthorized access difficult.

### Tied to Physical Devices

A passkey is bound to a specific device, meaning even if someone knows your credentials, they cannot log in without your device.

### Resistant to Data Breaches
Since only the public key is stored on the server, even if the server is breached, the stolen data is useless without the corresponding private key.

## Passkey extension for Magento 2
The Passkey Extension for Magento 2, enables administrators to log in using a passkey. It enhances the user experience
by simplifying the login process while maintaining high-security standards. This is ideal for shop administrators who
wish to log in without a password while still ensuring security.

#### Key Features:

1. **Easy Login:** By Passkey you can log in without using your username and password.

> 2. **Secure:** Passkey is a secure way to log in without using your username and password.

Reviewer: That is mainly the same point as above ("without using your username and password"). Can we change that somehow?

5. `bin/magento setup:upgrade`
6. `bin/magento setup:di:compile`
7. `bin/magento cache:flush`

## Configuration
You can enable the Passkey feature in the Magento backend under `Stores > Configuration > Security > 2FA > General`.
It is important to mention that the Passkey can not be the only 2FA method enabled. You need to enable at least one other 2FA method.
If you just choose Passkey as the only 2FA method, you will get an error message.

![Settings configuration]({{ "images/passkey/passkey-configuration.png" }} "Settings configuration")

## Usage
After enabling Passkey, you need to register a device to use it, such as a mobile phone, tablet, or YubiKey. To register a new device (after enabling Passkey 2FA), first log out, then log in again using your username and password.

![Passkey device registration Email]({{ "images/passkey/passkey-device-registration-email.png" }} "Passkey device registration Email")

Next, you will be prompted to check your email and click the link provided. Once you click the link, you'll be guided to register your Passkey device. Simply select the device you want to register and follow the on-screen instructions.

> for example to registering Yubi-Key, you need to insert the Yubi-Key into the USB port and touch the Yubi-Key.

Reviewer: ```suggestion
For example to register Yubi-Key, you need to insert the Yubi-Key into the USB port and touch it.
```

![Passkey device registration]({{ "images/passkey/passkey-register-device.png" }} "Passkey device registration")

After registering your device you will be redirected to the Admin Panel.

### Logging in with Passkey

> After registering your Passkey device, you can log in with Passkey. To do so, you just need to click on the Passkey button in the login form. and you of course you do not need to enter your username and password.

Reviewer: ```suggestion
After registering your Passkey device, you can log in with Passkey. To do so, you just need to click on the Passkey button in the login form. And you of course you do not need to enter your username and password anymore.
```

> ![Log in by passkey]({{ "images/passkey/login-by-passkey.png" }} "Log in by Passkey")

Reviewer: Could we use an image, without the "You have logged out" hint? This looks kind of strange, as we would like to login at this point.

> ### Resetting the Passkey

Reviewer: Please explain, that you need to have CLI server access to do this.
Admin users might not be familiar with that or might not have it.
Is there any other way of resetting it in the admin panel? Maybe with another logged in admin user?

---
title: Passkey (Magento 2)
permalink: passkey-m2.html

> summary: "The Passkey Extension for Magento 2 enables administrators to log in using a passkey"

Reviewer: ```suggestion
summary: "The Passkey Extension for Magento 2 enables administrators to log in using a Passkey"
```

![Passkey device registration Email]({{ "images/passkey/passkey-device-registration-email.png" }} "Passkey device registration Email")

Next, you will be prompted to check your email and click the link provided. Once you click the link, you'll be guided to register your Passkey device. Simply select the device you want to register and follow the on-screen instructions.
For example to register Yubi-Key, you need to insert the Yubi-Key into the USB port and touch it.

![Passkey device registration]({{ "images/passkey/passkey-register-device.png" }} "Passkey device registration")

After registering your device you will be redirected to the Admin Panel.

### Logging in with Passkey
After registering your Passkey device, you can log in with Passkey. To do so, you just need to click on the Passkey button in the login form. And you of course you do not need to enter your username and password anymore.

![Log in by passkey]({{ "images/passkey/login-by-passkey.png" }} "Log in by Passkey")

### Resetting the Passkey

> There are two ways to reset the Passkey.

Reviewer: ```suggestion
Imagine you lost your Passkey device and need to register a new one. There are two ways to reset the Passkey.
```

> #### Reset the Passkey in the Admin Panel

Reviewer: Awesome, that is a lot better

1. On the Admin sidebar, go to System > Permissions > All Users.
2. Select the user and open the account in edit mode.
3. Scroll down to the Current User Identity Verification section and enter your password.
4. In the left panel, click 2FA.
5. In the Configuration reset section, click Reset passkey and OK to confirm.

![Reset Passkey]({{ "images/passkey/reset-passkey.png" }} "Reset Passkey")

#### Reset the Passkey via CLI
For resetting the passkey with CLI you need to have access to the command line of the server, where your Magento installation is placed. If you don't have access to it, please ask your developer or your agency for help.

> Imagine you lost your Passkey device, you can reset your Passkey for any specific user by running the following command:

Reviewer: ```suggestion
There, you can reset your Passkey for any specific user by running the following command:
```
